ios/UARCTModule/UARCTEventEmitter.m

```diff
@@ -90,14 +95,22 @@ -(void)deepLinkReceived:(NSString *)deepLink {
 -(void)receivedForegroundNotification:(UANotificationContent *)notificationContent
                     completionHandler:(void (^)(void))completionHandler {
 
-    [self sendEventWithName:UARCTPushReceivedEventName body:[UARCTEventEmitter eventBodyForNotificationContent:notificationContent]];
+    NSDictionary *body = [UARCTEventEmitter eventBodyForNotificationContent:notificationContent];
+
+    if (![self sendEventWithName:UARCTPushReceivedEventName body:body]) {
+        [self.pendingEvents addObject:@{ @"name": UARCTPushReceivedEventName, @"body": body }];
+    }
     completionHandler();
 }
 
 -(void)receivedBackgroundNotification:(UANotificationContent *)notificationContent
                     completionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
 
-    [self sendEventWithName:UARCTPushReceivedEventName body:[UARCTEventEmitter eventBodyForNotificationContent:notificationContent]];
+    NSDictionary *body = [UARCTEventEmitter eventBodyForNotificationContent:notificationContent];
+
+    if (![self sendEventWithName:UARCTPushReceivedEventName body:body]) {
```

Comment:
is there any event left that will get dropped on the ground? Might be able to move the add to pending events logic into sendEventWithName:body:

Reply:
Makes sense, I can def change that

Reply:
Be careful here, the other method during `addListener:` might modify the array that you are iterating over. Probably want to make a copy of it in `addListener:` if you do move it into sendEventWIthName